source/Ship.cpp

```diff
@@ -3665,6 +3837,9 @@ void Ship::AddOutfit(const Outfit *outfit, int count)
 		// and cache this ship's jump range.
 		if(outfit->Get("jump drive"))
 			jumpRange = JumpRange(false);
+
+		// If any drone bay modifications were made then update ship drone bays.
+		UpdateBaysFromOutfits();
```

Comment:
This should probably be behind a conditional check if outfit is a weapon and is a bay.

Reply:
Conditional added